lib/cli/run-helpers.js

```diff
@@ -98,21 +98,29 @@ exports.handleRequires = (requires = []) => {
  * @param {boolean} [opts.exit] - Whether or not to force-exit after tests are complete
  * @param {Object} fileCollectParams - Parameters that control test
  *   file collection. See `lib/cli/collect-files.js`.
- * @returns {Runner}
+ * @returns {Promise<Runner>}
  * @private
  */
 exports.singleRun = (mocha, {exit}, fileCollectParams) => {
   const files = collectFiles(fileCollectParams);
   debug('running tests with files', files);
   mocha.files = files;
-  return mocha.run(exit ? exitMocha : exitMochaLater);
+
```

Comment:
My IDE tells me, `singleRun` could be converted to an async function. I think she is clever, this module is not exposed to our browser part. I leave it upto you.